src/main/java/Task.java

```diff
@@ -0,0 +1,76 @@
+import java.util.ArrayList;
+
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    protected String type; // T or D or E
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    public String toFileString() {
+        return null;
+    }
+
+    public static void addTask(String text, ArrayList<Task> tasksArrayList, int num) throws InvalidTodoSyntaxException {
+        // Todo
+        if (text.startsWith("todo")) {
```

Comment:
This logic can be abstracted to a parser class who will create the addTask command that can be executed which will call the addTask function in the tasklist class

Reply:
```
String fullCommand = ui.readCommand();
Command c = Parser.parseCommand(fullCommand);
c.execute(tasks, storage, ui);
```

something like this in main, can take reference from personsbook

Reply:
you can also consider keeping track of tasksArraylist with a class "TaskList", that contains the addTask method that adds task to the tasksArraylist